include/flash.h

```diff
@@ -127,6 +137,28 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+/**
+ *  @brief  Get layout of pages of flash.
```

Comment:
In addition to @mbolivar proposal I'd also like to have a routine to find sector boundaries given a sector index, i.e. a number in range [0..total_number_of_sectors).

The use case here is: I have a flash area and I need to find sectors which entirely fit inside so I can look through all sectors to find which are inside the area. In particular NFFS implementation (port in progress) does this to partition flash area into its own areas.